Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -400,60 +332,193 @@ else if (world.equals(craft.getWorld())
+                    && (w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType().isAir()
+                    || craft.getType().getMaterialSetProperty(CraftType.PASSTHROUGH_BLOCKS).contains(
+                    w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType()))) {
+                groundY--;
+            }
+
+            if (centreMinY - groundY > craft.getType().getIntProperty(CraftType.HOVER_LIMIT))
+                dy = -1;
+            return false;
+        }
+
+        dy = dropDistance(oldHitBox);
+        return false;
+    }
+
+    private void processAllNetherPortals() throws ExecutionException, InterruptedException {
+        // proccess nether portals
+        if (!Settings.CraftsUseNetherPortals
+                || craft.getWorld().getEnvironment() == Environment.THE_END
+                || !world.equals(craft.getWorld())) {
+            return;
+        }
+
+        // ensure chunks are loaded for portal checking only if change in location is
+        // large
+        Set<MovecraftChunk> chunksToLoad = ChunkManager.getChunks(oldHitBox, world, dx, dy, dz);
+        MovecraftChunk.addSurroundingChunks(chunksToLoad, 2);
+        ChunkManager.checkChunks(chunksToLoad);
+        if (!chunksToLoad.isEmpty())
+            ChunkManager.syncLoadChunks(chunksToLoad).get();
+
+        for (MovecraftLocation oldLocation : oldHitBox) {
+            Location location = oldLocation.translate(dx, dy, dz).toBukkit(craft.getWorld());
+            Block block = craft.getWorld().getBlockAt(location);
+
+            if (block.getType() != Material.NETHER_PORTAL) {
+                continue;
+            }
+
+            if (processNetherPortal(block)) {
+                sound = Sound.BLOCK_PORTAL_TRAVEL;
+                volume = 0.25f;
+                break;
+            }
+
+        }
+    }
+
+    private void processHighCraft(int minY, int maxY) {
+        // Only modify dy when not switching worlds
+        //Check if the craft is too high
+        boolean sameWorld = world.equals(craft.getWorld());
+        CraftType craftType = craft.getType();
+        if (!sameWorld) {
+            return;
+        }
+
+        if ((int) craftType.getPerWorldProperty(CraftType.PER_WORLD_MAX_HEIGHT_LIMIT, craft.getWorld()) < craft.getHitBox().getMinY()) {
+            dy = Math.min(dy, -1);
+            return;
+        }
+        final int perWorldMaxHeigh = (int) craftType.getPerWorldProperty(CraftType.PER_WORLD_MAX_HEIGHT_ABOVE_GROUND, world);
+        // world == craft.world
+        if (perWorldMaxHeigh > 0) {
+            final MovecraftLocation middle = oldHitBox.getMidPoint();
+            int testY = minY;
+            while (testY > 0) {
+                testY--;
+                if (!craft.getWorld().getBlockAt(middle.getX(), testY, middle.getZ()).getType().isAir())
+                    break;
+            }
+            if (maxY - testY > perWorldMaxHeigh)
+                dy = Math.min(dy, -1);
         }
```

Comment:
You can invert this if statement and return if height max height is less than or equal to world min height.